core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1247,7 +1252,7 @@ class DAGScheduler(
                 }
               }
 
-              // Note: newly runnable stages will be submitted below when we submit waiting stages
+              submitWaitingChildStages(shuffleStage)
```

Comment:
Should this be done when !shuffleStage.isAvailable and we have resubmitted the shuffleStage, or only within the else branch?